databases/jdp/database_test.go

```diff
@@ -0,0 +1,121 @@
+package jdp
+
+import (
+	"github.com/stretchr/testify/assert"
+	"os"
+	"testing"
+
+	"dts/config"
+	"dts/core"
+)
+
+const jdpConfig string = `
+databases:
+  jdp:
+    name: JGI Data Portal
+    organization: Joint Genome Institue
+    url: https://files.jgi.doe.gov
+    endpoint: globus-jdp
+    auth:
+      client_id: ${JGI_CLIENT_ID}
+      client_secret: ${JGI_CLIENT_SECRET}
+endpoints:
+  globus-jdp:
+    name: Globus NERSC DTN
+    id: ${DTS_GLOBUS_TEST_ENDPOINT}
+    provider: globus
+    auth:
+      client_id: ${DTS_GLOBUS_CLIENT_ID}
+      client_secret: ${DTS_GLOBUS_CLIENT_SECRET}
+`
+
+// this function gets called at the begіnning of a test session
+func setup() {
+	config.Init([]byte(jdpConfig))
+}
+
+// this function gets called after all tests have been run
+func breakdown() {
+}
+
+func TestNewDatabase(t *testing.T) {
+	assert := assert.New(t) // binds assert to t
+	orcid := os.Getenv("DTS_KBASE_TEST_ORCID")
+	jdpDb, err := NewDatabase(orcid)
+	assert.NotNil(jdpDb, "JDP database not created")
+	assert.Nil(err, "JDP database creation encountered an error")
+}
+
+func TestNewDatabaseWithoutOrcid(t *testing.T) {
+	assert := assert.New(t) // binds assert to t
+	jdpDb, err := NewDatabase("")
```

Comment:
Does go have the concept of null / undefined, and if so, is it worth adding tests where `orcid` is null/undef?

Reply:
Go does have `nil`, but it doesn't have `nil` strings (only blank ones).

Reply:
Btw, strings and "byte slices" (`[]byte`) are cousins in GoLand, and all slices do a have a `nil`able representation.